File: benchmarks/oopsla25/subjects/511.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f511(const char *s) {
    if (strcmp(s, "abc") == 0) {
        return;
    } else if (strcmp(s, "ab") == 0) {
        return;
    } else if (strcmp(s, "a") == 0) {
        return;
    } else if (strcmp(s, "ac") == 0) {
        return;
    } else if (strcmp(s, "bc") == 0) {
        return;
    } else if (strcmp(s, "b") == 0) {
        return;
    } else if (strcmp(s, "c") == 0) {
        return;
    } else {
        assert(strcmp(s, "") == 0);
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f511(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/511.py
import sys

def f511(s: str):
  if s == "abc":
    return
  elif s == "ab":
    return
  elif s == "a":
    return
  elif s == "ac":
    return
  elif s == "bc":
    return
  elif s == "b":
    return
  elif s == "c":
    return
  else:
    assert s == ""

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f511(my_string)
